documentation/modules/managing/con-consumer-config-properties.adoc

```diff
@@ -6,11 +6,18 @@
 = Kafka consumer configuration tuning
 
 [role="_abstract"]
-Use a basic consumer configuration with optional properties that are tailored to specific use cases.
-
+Use configuration properties to optimize the performance of Kafka consumers.
+You can use standard Kafka consumer configuration options.
 When tuning your consumers your primary concern will be ensuring that they cope efficiently with the amount of data ingested.
 As with the producer tuning, be prepared to make incremental changes until the consumers operate as expected.
 
+When tuning a consumer, consider the following aspects carefully, as they significantly impact its performance and behavior:
+
+Scaling:: Consumer groups enable parallel processing of messages by distributing the load across multiple consumers. Properly configuring consumer groups can enhance scalability and throughput.
+Message ordering:: A consumer observes messages in a single partition in the same order that they were committed to the broker, which means that Kafka only provides ordering guarantees for messages in a single partition. If absolute ordering within a topic is important, use a single-partition topic. While a single-partition topic allows only one consumer instance, you can leverage a pool of worker threads to create a multi-threaded consumer for more efficient processing. When processing messages for specific entities (like users), you can use the ID (like user ID) as the message key and direct all messages to a single partition. This way, you can maintain message ordering for events specific to individual entities. If a new entity is created, you can create a new topic and migrate its data.
```

Comment:
This is super hard to comment on because it is one long line. I would switch the first two sentences. Single partition topic allows only `one consumer instance to consumer from it at a point in time`. But you can have more of them. The threading suggestion seems wrong to me => it makes it sound super easy, but without individual-per-message acks it is super hard to implement. The users as the keys are also a bit oversimplified.

Reply:
- Switched the first two lines
- I think we can live without the threading part here -- removed
- Not sure how much more in-depth we should get for using keys here. I changed the text a bit here

> Message ordering:: 
> If absolute ordering within a topic is important, use a single-partition topic. 
> A consumer observes messages in a single partition in the same order that they were committed to the broker, which means that Kafka only provides ordering guarantees for messages in a single partition. 
> It is also possible to maintain message ordering for events specific to individual entities, such as users.
> If a new entity is created, you can create a new topic dedicated to that entity.
> You can use a unique ID, like a user ID, as the message key and route all messages with the same key to a single partition within the topic.